src/sidebar/util/annotation-sharing.js

```diff
@@ -1,47 +1,74 @@
 /**
+ * @typedef {import('../../types/api').Annotation} Annotation
  * @typedef {import('../../types/config').HostConfig} HostConfig
  */
 
 import serviceConfig from '../service-config';
 
 /**
- * Is the sharing of annotations enabled? Check for any defined `serviceConfig`,
- * but default to `true` if none found.
+ * Retrieve an appropriate sharing link for this annotation.
  *
- * @param {HostConfig} settings
- * @return {boolean}
+ * If the annotation is on a shareable document (i.e. its document is
+ * web-accessible), prefer the `incontext` (bouncer) link, but fallback to the
+ * `html` (single-annotation `h` web view) link if needed.
+ *
+ * If the annotation is not on a shareable document, don't use the `incontext`
+ * link as that won't work; only use the single-annotation-view `html` link.
```

Comment:
This comment raises an issue - does the backend even attempt to generate an `incontext` link for `file:` URLs? It probably shouldn't.

Reply:
IIRC, yes: it will always return an `incontext` (bouncer) link unless the `h` service is not configured with a `bouncer_url`.